src/component/MovieCard.js

```diff
@@ -4,23 +4,37 @@ import { MOVIE_GENRES } from '../constants/movies'
 import { withRouter } from 'react-router-dom';
 import MovieLikes from './movies/MovieLikes';
 
-const MovieCard = ({ movie, history }) => (
-  <div className="card">
-    <button className="btn p-0" onClick={ () => history.push('/movie/' + movie.id) }>
-      <div className="embed-responsive embed-responsive-4by3">
-        <img className="card-img-top embed-responsive-item" src={ movie.cover_image_url } alt="Not available :(" />
-      </div>
-      <div className="card-body pb-0">
-        <h4 className="card-title">{ movie.title }</h4>
-        <h6 className="card-subtitle mb-3 text-muted">{ MOVIE_GENRES[movie.genre] }</h6>
-        <p className="card-text">{ movie.description }</p>
-        <p className="card-text text-left text-muted">Visits: { movie.visits }</p>
-      </div>
-    </button>
-    <hr />
-    <div className="card-body p-0">
+const MovieCard = ({ movie, history, addMovieToWatchlist }) => (
+  <div className="card text-center">
+    <div className="embed-responsive embed-responsive-4by3">
+      <img className="card-img-top embed-responsive-item" src={ movie.cover_image_url } alt="Not available :(" />
+    </div>
+    <div className="card-body pb-0">
+      { movie.user_watched &&
+        <div className="d-flex justify-content-end pb-2">
+          <span className='badge badge-pill badge-success'>
+              Watched <i className="fa fa-check"></i>
+          </span>
+        </div>
+      }
+      <h4 className="card-title">{ movie.title }</h4>
+      <h6 className="card-subtitle mb-3 text-muted">{ MOVIE_GENRES[movie.genre] }</h6>
+      <p className="card-text">{ movie.description }</p>
+    </div>
+    <div className="card-body pb-0">
       <MovieLikes movie={ movie } />
     </div>
+    <div className="d-flex justify-content-between pl-3">
+        <p className="card-text text-muted my-auto">Visits: { movie.visits }</p>
+        { !movie.in_user_watch_list &&
+          <button className="btn text-primary" onClick={ () => addMovieToWatchlist(movie.id) }>
+            <i className="fa fa-plus mr-1"/> Add to watchlist
+          </button>
+        }
```

Comment:
ovdje ne bi bilo lose da korisniku ponudimo da ima kao neki toggle, tipa ako nije dodao da bude add to watchlist, a ako jeste ubacio da onda isto tako moze lako izbaciti sa `Remove from watchlist`, tkao nesto

Reply:
Ova funkcionalnost postoji, ali unutar same `Watchlist` komponente. Tamo imam dugme pomocu kog moze da obrise film iz watch liste.

Reply:
Dodao sam i u MovieCard i u MoviePage komponente tu funkcionalnost, jer na ta dva mesta postoji add to watchlist.

Reply:
Odradjeno

Reply:
Jasno, ali ideja je na licu mjesta da mozes odmah da skines, da ne moras ici na posebnu stranicu i da izbacis iz watchlist-e

Reply:
Shvatio sam da je bespotrebno terati korisnika da ide na drugu stranicu da bi uradio remove, tako da sam odradio to i na mestima gde imam add dakle i u movie card komponenti kao i u movie page.